File: solutions/python/src/charging/domain/id_tag.py
"""The credential presented to start charging (RFID card, app token): "TAG-MONTHLY-77"."""
from charging.domain.errors import DomainError


def parse_id_tag(raw: str) -> str:
    value = raw.strip()
    if not value or len(value) > 20:
        raise DomainError(f'Invalid idTag: "{raw}"')
    return value

File: solutions/node/src/charging/domain/IdTag.ts
import { DomainError } from './errors.ts';

// The credential presented to start charging (RFID card, app token). "TAG-MONTHLY-77".
export type IdTag = string;

export function parseIdTag(raw: string): IdTag {
  const value = raw.trim();
  if (value.length === 0 || value.length > 20) throw new DomainError(`Invalid idTag: "${raw}"`);
  return value;
}
